### DISCUSSION TITLE
[@types/selenium-webdriver] is too outdated

### DISCUSSION THREAD

**Discussion by andrew-aladev** [DefinitelyTyped/DefinitelyTyped#65762] [Issues with a @types package]
It looks like current `@types/selenium-webdriver` is too outdated for any reasonable usage. For example [input.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/selenium-webdriver/lib/input.d.ts) should reflect [input.js](https://github.com/SeleniumHQ/selenium/blob/trunk/javascript/node/selenium-webdriver/lib/input.js), but is actually reflects just [4.1.0 version](https://github.com/SeleniumHQ/selenium/blob/selenium-4.1.0/javascript/node/selenium-webdriver/lib/input.js).

Latest version of selenium input includes more buttons, scrolling, more options for actions, etc. Current version is not usable. If you won't plan to update it than i think it will be better to remove it completely from definitely typed. Now this ancient version just confuses developers.

[a comment by an automated account is not shown]

**Comment by andrew-aladev** [comment]
You can just do:

```
git diff selenium-4.1.0..selenium-4.10.0 -- javascript/node/selenium-webdriver/lib
```

You can see how much is missing and it is just core functionality.

**Comment by SupernaviX** [comment]
A long time ago, I made a PR to update the typings for selenium v3 . That was back when I was a node developer who used selenium at my day job, which isn't the case anymore. I'm not affiliated with selenium at all, and I'm not sure if any of the other contributors are/were.

I've reviewed a bunch of PRs for the webdriver typings, though I haven't actually contributed myself for a long time. There doesn't seem to be any one "maintainer" for it, people normally contribute one-off things.

If you've got knowledge of typescript and recent selenium versions, I'd recommend opening a PR/PRs with any missing bits you want to use. I'll review it :smile:
